geo/geomfn: implement ST_SymmetricDifference

CyrusJavan opened the pull request:
Added ST_SymmetricDifference builtin.

ST_SymmetricDifference is simply an alias for ST_SymDifference.

Release justification: low-risk update to new functionality
Release note (sql change): Implemented the geometry based builtin
`ST_SymmetricDifference`.

Resolves: #49052

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53688)
<!-- Reviewable:end -->

otan commented on pkg/sql/logictest/testdata/logic_test/geospatial:
```diff
@@ -1013,161 +1013,162 @@ SELECT ST_IsRing('POINT(0 0)')
   ST_Intersection(a.geom, b.geom) = ST_Intersection(a.geom::string, b.geom::string),
-  ST_AsEWKT(ST_SymDifference(a.geom, b.geom))
+  ST_AsEWKT(ST_SymDifference(a.geom, b.geom)),
+  ST_AsEWKT(ST_SymmetricDifference(a.geom, b.geom))
```
thanks! quick comment -- can you move both these symdifference statements to a separate select?

CyrusJavan replied:
Done 👍

CyrusJavan committed:
geo/geomfn: implement ST_SymmetricDifference

Added ST_SymmetricDifference builtin.

ST_SymmetricDifference is simply an alias for ST_SymDifference.

Release justification: low-risk update to new functionality
Release note (sql change): Implemented the geometry based builtin .

Resolves: #49052

otan commented:
bors r+

thanks!
